app/components/primer/select_menu_component.rb

```diff
@@ -0,0 +1,499 @@
+# frozen_string_literal: true
+
+module Primer
+  # Use select menus to list clickable choices, allow filtering between them, and highlight
+  # which ones are selected.
+  class SelectMenuComponent < Primer::Component
+    include ViewComponent::Slotable
+
+    LIST_BORDER_CLASSES = {
+      all: nil,
+      omit_top: "border-top-0",
+      none: "SelectMenu-list--borderless",
+    }.freeze
+    DEFAULT_LIST_BORDER_CLASS = :all
+    DEFAULT_LOADING = false
+    DEFAULT_BLANKSLATE = false
+    DEFAULT_ALIGN_RIGHT = false
+
+    with_slot :header, class_name: "Header"
+    with_slot :item, class_name: "Item", collection: true
+    with_slot :tab, class_name: "Tab", collection: true
+    with_slot :filter, class_name: "Filter"
+    with_slot :footer, class_name: "Footer"
+
+    attr_reader :message
+
+    #
+    # @example 193|Basic example|Use a `DetailsComponent` to toggle the select menu, with the `body` of the details component holding the select menu.
+    #   <%= render Primer::DetailsComponent.new(overlay: :default, reset: true, position: :relative) do |details_component| %>
```

Comment:
I'm wondering if SelectMenu should handle the DetailsComponent implementation as well, as the React SelectMenu effectively does. What do you think?

Reply:
I would be all right with that. It felt odd having the `omit_wrapper` argument.

Reply:
Working on this!